### ISSUE TITLE
Request support for TH01CB3S // TH01 CB3S

### ISSUE THREAD

**Issue by fireheadman** [make-all/tuya-local#1270]
This is a Temperature and Humidity device --> wifi connected with I believe tuyaMCU.
Logs are provided from the TUYA Integration device.

# Log Message

```
Logger: custom_components.tuya_local.config_flow
Source: custom_components/tuya_local/config_flow.py:216
Integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: 10:10:08 PM (1 occurrences)
Last logged: 10:10:08 PM

Connection test failed with <class 'Exception'> Unable to find device on network (specify IP address)

{
  "home_assistant": {
    "installation_type": "Home Assistant Supervised",
    "version": "2023.11.1",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.6",
    "docker": true,
    "arch": "x86_64",
    "timezone": "America/Denver",
    "os_name": "Linux",
    "os_version": "6.1.0-13-amd64",
    "supervisor": "2023.10.1",
    "host_os": "Debian GNU/Linux 12 (bookworm)",
    "docker_version": "24.0.7",
    "chassis": "desktop",
    "run_as_root": true
  },
  "custom_components": {
    "tuya_local": {
      "version": "2023.9.2",
      "requirements": [
        "pycryptodome~=3.18",
        "tinytuya==1.12.11"
      ]
    }
  },
  "integration_manifest": {
    "domain": "tuya",
    "name": "Tuya",
    "codeowners": [
      "@Tuya",
      "@zlinoliver",
      "@frenck"
    ],
    "config_flow": true,
    "dependencies": [
      "ffmpeg"
    ],
    "dhcp": [
      {
        "macaddress": "105A17*"
      },
      {
        "macaddress": "10D561*"
      },
      {
        "macaddress": "1869D8*"
      },
      {
        "macaddress": "381F8D*"
      },
      {
        "macaddress": "508A06*"
      },
      {
        "macaddress": "68572D*"
      },
      {
        "macaddress": "708976*"
      },
      {
        "macaddress": "7CF666*"
      },
      {
        "macaddress": "84E342*"
      },
      {
        "macaddress": "D4A651*"
      },
      {
        "macaddress": "D81F12*"
      }
    ],
    "documentation": "https://www.home-assistant.io/integrations/tuya",
    "integration_type": "hub",
    "iot_class": "cloud_push",
    "loggers": [
      "tuya_iot"
    ],
    "requirements": [
      "tuya-iot-py-sdk==0.6.6"
    ],
    "is_built_in": true
  },
  "data": {
    "endpoint": "https://openapi.tuyaus.com",
    "auth_type": 0,
    "country_code": "1",
    "app_type": "smartlife",
    "mqtt_connected": true,
    "disabled_by": null,
    "disabled_polling": false,
    "name": "T & H Sensor",
    "model": "TH01 CB3S",
    "category": "wsdcg",
    "product_id": "lf36y5nwb8jkxwgg",
    "product_name": "T & H Sensor",
    "online": true,
    "sub": false,
    "time_zone": "-06:00",
    "active_time": "2023-09-30T02:16:05+00:00",
    "create_time": "2023-09-30T02:16:05+00:00",
    "update_time": "2023-11-08T15:26:55+00:00",
    "function": {
      "temp_unit_convert": {
        "type": "Enum",
        "value": {
          "range": [
            "c",
            "f"
          ]
        }
      }
    },
    "status_range": {
      "va_temperature": {
        "type": "Integer",
        "value": {
          "unit": "\u2103",
          "min": -200,
          "max": 600,
          "scale": 1,
          "step": 1
        }
      },
      "va_humidity": {
        "type": "Integer",
        "value": {
          "unit": "%",
          "min": 0,
          "max": 99,
          "scale": 0,
          "step": 1
        }
      },
      "battery_state": {
        "type": "Enum",
        "value": {
          "range": [
            "low",
            "middle",
            "high"
          ]
        }
      },
      "temp_unit_convert": {
        "type": "Enum",
        "value": {
          "range": [
            "c",
            "f"
          ]
        }
      }
    },
    "status": {
      "va_temperature": 122,
      "va_humidity": 39,
      "battery_state": "high",
      "temp_unit_convert": "f"
    },
    "home_assistant": {
      "name": "T & H Sensor",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "temperature",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": "\u00b0F",
          "state": {
            "entity_id": "sensor.t_h_sensor_temperature",
            "state": "54.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "\u00b0F",
              "device_class": "temperature",
              "friendly_name": "T & H Sensor Temperature"
            },
            "last_changed": "2023-11-08T15:52:25.098194+00:00",
            "last_updated": "2023-11-08T15:52:25.098194+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "humidity",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": "%",
          "state": {
            "entity_id": "sensor.t_h_sensor_humidity",
            "state": "39.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "%",
              "device_class": "humidity",
              "friendly_name": "T & H Sensor Humidity"
            },
            "last_changed": "2023-11-08T15:52:25.099713+00:00",
            "last_updated": "2023-11-08T15:52:25.099713+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "original_icon": "mdi:battery",
          "unit_of_measurement": null,
          "state": {
            "entity_id": "sensor.t_h_sensor_battery_state",
            "state": "high",
            "attributes": {
              "icon": "mdi:battery",
              "friendly_name": "T & H Sensor Battery state"
            },
            "last_changed": "2023-11-08T15:52:25.100519+00:00",
            "last_updated": "2023-11-08T15:52:25.100519+00:00"
          }
        }
      ]
    }
  }
}
```

# Information about DPS mappings

```
{"1":"Temperature","2":"Humidity","3":"Battery level state","9":"Temp unit convert"}

{
  "result": {
    "model": "{\"modelId\":\"000004ax7p\",\"services\":[{\"actions\":[],\"code\":\"\",\"description\":\"\",\"events\":[],\"name\":\"默认服务\",\"properties\":[{\"abilityId\":1,\"accessMode\":\"ro\",\"code\":\"temp_current\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-dp_c\",\"attribute\":\"1666\",\"trigger\":\"direct\"},\"name\":\"温度\",\"typeSpec\":{\"max\":600,\"min\":-200,\"scale\":1,\"step\":1,\"type\":\"value\",\"typeDefaultValue\":-200,\"unit\":\"℃\"}},{\"abilityId\":2,\"accessMode\":\"ro\",\"code\":\"humidity_value\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-shidu\",\"attribute\":\"1666\",\"trigger\":\"direct\"},\"name\":\"湿度\",\"typeSpec\":{\"max\":99,\"min\":0,\"scale\":0,\"step\":1,\"type\":\"value\",\"typeDefaultValue\":0,\"unit\":\"%\"}},{\"abilityId\":3,\"accessMode\":\"ro\",\"code\":\"battery_state\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-dp_battery\",\"attribute\":\"1152\",\"trigger\":\"direct\"},\"name\":\"电池电量状态\",\"typeSpec\":{\"range\":[\"low\",\"middle\",\"high\"],\"type\":\"enum\",\"typeDefaultValue\":\"low\"}},{\"abilityId\":9,\"accessMode\":\"rw\",\"code\":\"temp_unit_convert\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-dp_mode\",\"attribute\":\"1152\"},\"name\":\"温标切换\",\"typeSpec\":{\"range\":[\"c\",\"f\"],\"type\":\"enum\",\"typeDefaultValue\":\"c\"}}]}]}"
  },
  "success": true,
  "t": 1699852075413,
  "tid": "9ab1c8d981e211ee8088063dea7c5c23"
}
```

# Product ID

    "product_id": "lf36y5nwb8jkxwgg",
    "product_name": "TH01CB3S"

Purchased it from amz.
[https://www.amazon.com/dp/B0B58VSJDG](url)

The default tuya integretion will pick this up fine.

![image](https://github.com/make-all/tuya-local/assets/19847195/95a8b1c5-6ce6-4c2b-8a0b-6812f9cfaa8b)
![image](https://github.com/make-all/tuya-local/assets/19847195/0f8d8a19-9ff7-4e4e-8e19-6a5c67d1d4df)

**Comment by fireheadman** [comment]
unsure how to get logs and dps info....
when I attempt to add it, and tried each version, nothing is outputted into HA logs.

I'm willing to help where ever possible, just let me know.

**Comment by fireheadman** [comment]
This is what it looks like when added from tuya integretion
![image](https://github.com/make-all/tuya-local/assets/19847195/64061301-ec63-4a8c-961a-606fe93866fa)

```
2023-11-08 08:52:25.097 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.tuya entity: sensor.t_h_sensor_temperature
2023-11-08 08:52:25.098 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.tuya entity: sensor.t_h_sensor_humidity
2023-11-08 08:52:25.100 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.tuya entity: sensor.t_h_sensor_battery_state
```

```
{
  "home_assistant": {
    "installation_type": "Home Assistant Supervised",
    "version": "2023.11.1",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.6",
    "docker": true,
    "arch": "x86_64",
    "timezone": "America/Denver",
    "os_name": "Linux",
    "os_version": "6.1.0-13-amd64",
    "supervisor": "2023.10.1",
    "host_os": "Debian GNU/Linux 12 (bookworm)",
    "docker_version": "24.0.7",
    "chassis": "desktop",
    "run_as_root": true
  },
  "custom_components": {
    "tuya_local": {
      "version": "2023.9.2",
      "requirements": [
        "pycryptodome~=3.18",
        "tinytuya==1.12.11"
      ]
    }
  },
  "integration_manifest": {
    "domain": "tuya",
    "name": "Tuya",
    "codeowners": [
      "@Tuya",
      "@zlinoliver",
      "@frenck"
    ],
    "config_flow": true,
    "dependencies": [
      "ffmpeg"
    ],
    "dhcp": [
      {
        "macaddress": "105A17*"
      },
      {
        "macaddress": "10D561*"
      },
      {
        "macaddress": "1869D8*"
      },
      {
        "macaddress": "381F8D*"
      },
      {
        "macaddress": "508A06*"
      },
      {
        "macaddress": "68572D*"
      },
      {
        "macaddress": "708976*"
      },
      {
        "macaddress": "7CF666*"
      },
      {
        "macaddress": "84E342*"
      },
      {
        "macaddress": "D4A651*"
      },
      {
        "macaddress": "D81F12*"
      }
    ],
    "documentation": "https://www.home-assistant.io/integrations/tuya",
    "integration_type": "hub",
    "iot_class": "cloud_push",
    "loggers": [
      "tuya_iot"
    ],
    "requirements": [
      "tuya-iot-py-sdk==0.6.6"
    ],
    "is_built_in": true
  },
  "data": {
    "endpoint": "https://openapi.tuyaus.com",
    "auth_type": 0,
    "country_code": "1",
    "app_type": "smartlife",
    "mqtt_connected": true,
    "disabled_by": null,
    "disabled_polling": false,
    "name": "T & H Sensor",
    "model": "TH01 CB3S",
    "category": "wsdcg",
    "product_id": "lf36y5nwb8jkxwgg",
    "product_name": "T & H Sensor",
    "online": true,
    "sub": false,
    "time_zone": "-06:00",
    "active_time": "2023-09-30T02:16:05+00:00",
    "create_time": "2023-09-30T02:16:05+00:00",
    "update_time": "2023-11-08T15:26:55+00:00",
    "function": {
      "temp_unit_convert": {
        "type": "Enum",
        "value": {
          "range": [
            "c",
            "f"
          ]
        }
      }
    },
    "status_range": {
      "va_temperature": {
        "type": "Integer",
        "value": {
          "unit": "\u2103",
          "min": -200,
          "max": 600,
          "scale": 1,
          "step": 1
        }
      },
      "va_humidity": {
        "type": "Integer",
        "value": {
          "unit": "%",
          "min": 0,
          "max": 99,
          "scale": 0,
          "step": 1
        }
      },
      "battery_state": {
        "type": "Enum",
        "value": {
          "range": [
            "low",
            "middle",
            "high"
          ]
        }
      },
      "temp_unit_convert": {
        "type": "Enum",
        "value": {
          "range": [
            "c",
            "f"
          ]
        }
      }
    },
    "status": {
      "va_temperature": 122,
      "va_humidity": 39,
      "battery_state": "high",
      "temp_unit_convert": "f"
    },
    "home_assistant": {
      "name": "T & H Sensor",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "temperature",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": "\u00b0F",
          "state": {
            "entity_id": "sensor.t_h_sensor_temperature",
            "state": "54.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "\u00b0F",
              "device_class": "temperature",
              "friendly_name": "T & H Sensor Temperature"
            },
            "last_changed": "2023-11-08T15:52:25.098194+00:00",
            "last_updated": "2023-11-08T15:52:25.098194+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": "humidity",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": "%",
          "state": {
            "entity_id": "sensor.t_h_sensor_humidity",
            "state": "39.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "%",
              "device_class": "humidity",
              "friendly_name": "T & H Sensor Humidity"
            },
            "last_changed": "2023-11-08T15:52:25.099713+00:00",
            "last_updated": "2023-11-08T15:52:25.099713+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "original_icon": "mdi:battery",
          "unit_of_measurement": null,
          "state": {
            "entity_id": "sensor.t_h_sensor_battery_state",
            "state": "high",
            "attributes": {
              "icon": "mdi:battery",
              "friendly_name": "T & H Sensor Battery state"
            },
            "last_changed": "2023-11-08T15:52:25.100519+00:00",
            "last_updated": "2023-11-08T15:52:25.100519+00:00"
          }
        }
      ]
    }
  }
}
```

**Comment by fireheadman** [comment]
Believe I understand more about this device and why it will not add...
Appears to be a low power device that activates its wifi beacon periodically to send data to tuya cloud.
I would need to time this to when it was pinging and attempt to add it.

I was also able to provide additional information on the Query Things Data Model for you....

<img width="387" alt="image" src="https://github.com/make-all/tuya-local/assets/19847195/49051ed5-584d-474e-932d-90ff869d7650">

**Comment by fireheadman** [comment]
Since this is completed, does that mean it is already in tuya-local?

**Comment by make-all** [comment]
No, it means this is out of scope for this integration

**Comment by fireheadman** [comment]
so I understand the scope of your integration, what makes it fall out of scope?

**Comment by make-all** [comment]
The fact it is battery powered makes it impossible to keep a connection to the device locally.  The device is designed to go into deep sleep and wake only when it wants to send something to the cloud.  If it was zigbee or bluetooth connected, there would be a hub in the middle we could keep connected to to ensure we get all the updates, but as a WiFi device, that is not an option.

**Comment by rayxxxxx** [comment]
@make-all I've got two of this devices: one connected with Tuya Local and the other one no because Tuya Local doesn't recognise the correct device type (why is a mystery). When the device is into sleep mode my sensor is unavailable but when it sends data the sensor get the info (I used a trigger based template sensor to have always an available sensor). It would be nice to have this integration for these devices.
